PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -71,7 +68,11 @@ class SearchViewModel: ObservableObject {
         self.userDefaults = userDefaults
         self.source = source
         networkPathMonitor.start(queue: .global())
-        self.searchService = source.makeSearchService()
+
+        self.savesLocalSearch = SavesLocalSearch(source: source)
+        self.savesOnlineSearch = SavesOnlineSearch(source: source)
+        self.archiveOnlineSearch = ArchiveOnlineSearch(source: source)
+        self.allOnlineSearch = AllOnlineSearch(source: source)
```

Comment:
is this preferred? I prefer the following to avoid duplicate code and still allows separate instances of the object so that caching is scope specific. Let me know if I a missing something.
```
self.savesOnlineSearch = OnlineSearch(source: source, scope: .saves)
 self.archiveOnlineSearch = OnlineSearch(source: source, scope: .archive)
 self.allOnlineSearch = OnlineSearch(source: source, scope: .all)
```

Reply:
I don't quite understand what the difference is.  You're going to end up with `OnlineSearch` and `OfflineSearch`. So it'll end up being the same amount of work?

Also, in this case, why are we using self?  I don't see a need for self to be used.

Reply:
One can get caught up in initializers doing that "unnecessarily". Typically, it's used because an initializer argument label matches that of a property name for the type being initializes, which makes it necessary. In this case, there's nothing matching, so it can be omitted.

Reply:
Discussed on Zoom:
* Will update to use to using this convention: https://github.com/kodecocodes/swift-style-guide#use-of-self
* Will update to using one class type for the Online searches